src/datta_lab_to_nwb/markowitz_gillis_nature_2023/markowitz_gillis_nature_2023behaviorinterface.py

```diff
@@ -1,22 +1,108 @@
 """Primary class for converting experiment-specific behavior."""
-from pynwb.file import NWBFile
-
+import numpy as np
+import pandas as pd
+from pynwb import NWBFile, TimeSeries
+from pynwb.behavior import (
+    BehavioralTimeSeries,
+    CompassDirection,
+    Position,
+    SpatialSeries,
+)
 from neuroconv.basedatainterface import BaseDataInterface
 
 class MarkowitzGillisNature2023BehaviorInterface(BaseDataInterface):
     """Behavior interface for markowitz_gillis_nature_2023 conversion"""
 
-    def __init__(self):
+    def __init__(self, filename: str):
         # This should load the data lazily and prepare variables you need
-        pass
+        columns = (
+            'predicted_syllable (offline)',
+            'uuid',
+            'date',
+            'session_name',
+            'SessionName',
+            'mouse_id',
+            'centroid_x_mm',
+            'centroid_y_mm',
+            'height_ave_mm',
+            'angle_unwrapped',
+            'timestamp',
+        )
+        metadata_columns = ["uuid", "date", "mouse_id"]
+        super().__init__(filename=filename, columns=columns, metadata_columns=metadata_columns)
 
-    def get_metadata(self):
-        # Automatically retrieve as much metadata as possible
+    def get_metadata(self, session_uuid: str) -> dict:
+        """Get metadata about the experiment as a whole."""
+        # TODO: store metadata in .yaml file
         metadata = super().get_metadata()
+
+        # get session metadata TODO: move session metadata to a separate file to avoid multiple reads
+        session_df = pd.read_parquet(self.source_data['filename'],
+                                     columns=self.source_data['columns'],
+                                     filters=[('uuid', '==', session_uuid)])
+        for col in self.source_data['metadata_columns']:
+            first_notnull = session_df.loc[session_df[col].notnull(), col].iloc[0]
+            metadata[col] = first_notnull
+        session_name = (
+                set(session_df.session_name[session_df.session_name.notnull()]) |
+                set(session_df.SessionName[session_df.SessionName.notnull()])
+                        )
+        assert len(session_name) == 1, "Multiple session names found"
+        metadata['session_name'] = session_name.pop()
```

Comment:
The general structures of the `metadata` structure of NeuroConv is as follows: `metadata[<modality>][<specific interface or reference to resulting neurodata type>]`

In particular, there is a special level called `NWBFile`; though there is no specific field for `session_name`

I would map this as `metadata["NWBFile"]["session_id"]`